api/src/main/java/uk/gov/dvsa/mot/trade/read/core/MotrReadServiceDatabase.java

```diff
@@ -35,11 +35,22 @@ public MotrReadServiceDatabase(MotTestReadService motTestReadService, VehicleRea
     @ProvideDbConnection
     public MotrResponse getLatestMotTestByRegistration(String registration) {
 
+        return getLatestMotTestByRegistration(registration, true);
+    }
+
+    @Override
+    @ProvideDbConnection
+    public MotrResponse getLatestMotTestByRegistration(String registration, Boolean withDvlaVehicles) {
+
         List<Vehicle> vehicles = vehicleReadService.findByRegistration(registration);
         MotrReadServiceDatabase.VehicleAndLatestMot vehicleAndLatestMot = getVehicleAndLatestMotTestPass(vehicles);
 
         if (vehicleAndLatestMot == null || !vehicleAndLatestMot.hasMotTest()) {
-            return getDvlaVehicleByRegistration(registration);
```

Comment:
let's split that one. DvlaVehicle does not have to have MotTest at all so logically it is hard to merge those two. Boolean flags as parameters to the method - noooo :-)